buildman/orchestrator.py

```diff
@@ -784,10 +808,17 @@ def _publish(self, **kwargs):
     def delete_key(self, key):
         assert not self.is_canceller_only
 
-        value = yield From(self._client.get(key))
-        yield From(self._client.delete(key))
-        yield From(self._client.delete(slash_join(key, REDIS_EXPIRING_SUFFIX)))
-        yield From(self._publish(event=KeyEvent.DELETE, key=key, value=value))
+        try:
+            value = yield From(self._client.get(key))
+            if value is None:
+                raise KeyError
```

Comment:
Why this change?

Reply:
From the interface:
>    Missing keys should return KeyError; otherwise, errors should raise an
    OrchestratorError.